Models/Queue.js

```diff
@@ -326,39 +397,55 @@ export class Queue {
       let jobData = JSON.parse(job.data);
       const errorMessage = error?.message || '';
 
-      this.realm.write(() => {
-        // Increment failed attempts number
-        if (!jobData.failedAttempts) {
-          jobData.failedAttempts = 1;
-        } else {
-          jobData.failedAttempts++;
-        }
-
-        // Log error
-        if (!jobData.errors) {
-          jobData.errors = [ errorMessage ];
-        } else {
-          jobData.errors.push(errorMessage);
-        }
-
-        job.data = JSON.stringify(jobData);
-
-        // Reset active status
-        job.active = false;
-
-        // Mark job as failed if too many attempts
-        if (jobData.failedAttempts >= jobData.attempts) {
-          job.failed = new Date();
-        }
-      });
-
-      // Execute job onFailure lifecycle callback.
-      this.worker.executeJobLifecycleCallback('onFailure', jobName, jobId, jobPayload);
-
-      // If job has failed all attempts execute job onFailed and onComplete lifecycle callbacks.
-      if (jobData.failedAttempts >= jobData.attempts) {
-        this.worker.executeJobLifecycleCallback('onFailed', jobName, jobId, jobPayload);
-        this.worker.executeJobLifecycleCallback('onComplete', jobName, jobId, jobPayload);
+      // Call the optional error profiler from the worker.options to learn what we should
+      // do with this error. If the profiler returns true, we should attempt the job.
+      const failureBehavior = this.worker.getFailureBehavior(jobName);
+
+      switch (failureBehavior) {
+        case 'standard':
+          this.realm.write(() => {
```

Comment:
break the inside of the case statement into a function so it's easier to validate the syntax of the swtich statement and easier to read